Valid streamlines projection and uniform scaling

carlOSx64 opened the pull request:
- Slices are scaled uniformly based on how the largest dimension fits into the destination size.

- Respects view orientation; flips image horizontally when necessary.

  - Before ![before_sheet](https://github.com/user-attachments/assets/76d54d52-77a6-4df0-bde3-c35d3c3652f4)
  - After ![after_sheet](https://github.com/user-attachments/assets/1feff039-d139-488f-990b-5cd6177d5d5b)

- Simplified scaling code using linear algebra; moved code out of `buffer.rs` into `resources.rs` and `workload.rs`.

- The fibers are now aligned with the slice correctly, and the rotation is applied. These images match the visualization in MI-Brain.
  - ```
    cargo run --release -- --views left right anterior posterior superior inferior -f testdata/Bundles/CC4_AnteriorMidBody.trk testdata/Anatomy/Subject__t1_brain_on_b0.nii.gz fibers/
    ```
    ![fiber_sheet](https://github.com/user-attachments/assets/0d88d617-c77b-4789-9474-da9326ef11df)

- Background now uses a clear color instead of clamping; no more color limitation and blurry border effect.

- Upgraded `glam` to `0.29`. Might remove it in the future and use `nalgebra` exclusively instead.

carlOSx64 committed:
Scaling uniformity and correct projection alignment

carlOSx64 committed:
Parametrized clear color, fixed axis mix-up, and simplified transform application

carlOSx64 committed:
Roadmap update

nilgoyette commented on src/main.rs:
```diff
@@ -81,6 +85,15 @@ fn main() {
+        .expect("The NIfTI must have consistent dimensions");
+
+    let dim: Vec<u32> = dim.iter().map(|&d| d as u32).collect();
+    UVec3::from_array(dim.try_into().expect("A 3D image is expected."))
```
You could directly create `UVec3` from `dim[0] as u32`, etc. instead of allocating. Maybe add an if before to check the length.

nilgoyette commented on src/slicer.rs:
```diff
@@ -16,11 +20,32 @@ pub enum Axis {
 pub enum View {
     Left,
-    Right, // TODO Switch camera
+    Right,
```
Hahaha, noooo, did you fix this?

carlOSx64 replied:
Yeah ! This is handled by the new `orientation()` method.

nilgoyette reviewed (commented):
Men, I feel like an imposter reading this code. I wouldn't even know where to start if I had to fix this. Bravo!

carlOSx64 committed:
Added parameters module, simplied transformations even further

carlOSx64 commented:
It's hard to code math without it being confusing, haha. I think I managed to simplify some aspects. I added a new `parameters` module that processes matrices which are the same for all slices and added some comments. I also redid the calculation for the `quad_transform`, so it's more straightforward now.

carlOSx64 committed:
Addressed review, minor refactoring

nilgoyette commented:
Thank you! I'll try to test with other datasets in the next days.
